app/src/main/java/com/android/bookswap/ui/chat/ChatScreen.kt

```diff
@@ -67,70 +85,184 @@ fun ChatScreen(
-                  Text("Send")
-                }
-          }
+            }
+      }
+    }
+    if (!updateActive) {
+      selectedMessage?.let { message ->
+        Popup(
+            alignment = Alignment.Center,
+            onDismissRequest = {
+              if (!updateActive) {
+                selectedMessage = null
+              }
+            }) {
+              Column(
+                  modifier =
+                      Modifier.background(ColorVariable.Primary, shape = RoundedCornerShape(8.dp))
+                          .border(2.dp, ColorVariable.Accent, shape = RoundedCornerShape(8.dp))
+                          .padding(8.dp),
+                  verticalArrangement = Arrangement.Center,
+                  horizontalAlignment = Alignment.CenterHorizontally) {
+                    Button(
+                        onClick = {
+                          // Handle edit
+                          newMessageText = TextFieldValue(message.text)
+                          updateActive = true
+                        },
+                        modifier =
+                            Modifier.background(
+                                    ColorVariable.Primary, shape = RoundedCornerShape(50))
+                                .padding(8.dp)
+                                .testTag("editButton")) {
+                          Text("Edit")
+                        }
+                    Button(
+                        onClick = {
+                          // Handle delete
+                          selectedMessage?.let { message ->
+                            messageRepository.deleteMessage(
+                                message.id,
+                                { result ->
+                                  if (result.isSuccess) {
+                                    Log.d("ChatScreen", "Message deleted successfully")
+                                    selectedMessage = null
+                                  } else {
+                                    Log.e(
+                                        "ChatScreen",
+                                        "Failed to delete message: ${result.exceptionOrNull()?.message}")
+                                  }
+                                },
+                                context)
+                          }
+                        },
+                        modifier =
+                            Modifier.background(
+                                    ColorVariable.Primary, shape = RoundedCornerShape(50))
+                                .padding(8.dp)
+                                .testTag("deleteButton")) {
+                          Text("Delete")
+                        }
+                  }
+            }
+      }
     }
   }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun MessageItem(message: DataMessage, currentUserId: String) {
+fun MessageItem(message: DataMessage, currentUserId: String, onLongPress: () -> Unit) {

```

Comment:
Could be add   a second argument onClick , because it possible that the action  to click is different if it's long or short could be different

Reply:
There is just no action for a click, so it makes no sense to add it, if I did, it would be left empty anyway, additionally, a OnLongPress does not trigger with a simple click, so adding it is unnecessary